### ISSUE TITLE
eval() vs freeze_model(True)

### ISSUE THREAD

**Issue by seahou3e** [Adapter-Hub/adapter-transformers#324]
### Environment info

- `adapter-transformers` version: 3.0.0
- Platform: ubuntu 20.04
- Python version: 3.7
- PyTorch version (GPU?): 1.11.0+cu113
### Description
I tried to perform evaluation on dev set after training of each epoch. Before training, I set
`optimizer.zero_grad()`, `model.train_adapter(adapter_i)` and `model.set_active_adapters(adapter_i)`,
and switch to `eval` mode after training (before evaluation on dev set):
`model.freeze_model(True)` and `model.set_active_adapters(adapter_i)`.
I found that if I change `model.freeze_model(True)` to `model.eval()`, the result will be very different after the complete training process. I am aware that `model.eval()` will leave parameters of adapters as `requires_grad=True`. But I set `optimizer.zero_grad()` before training every time, based on my understanding, these two commands (`model.freeze_model(True)` or `model.eval()`) should not cause different gradient computations during training of next epoch. Therefore I am confused why these two commands (set during evaluation on dev set) cause different results?

**Comment by hSterz** [comment]
Hey, 
the command `model.freeze_model(True)` does freeze the weights of the model. Hence, the weights are not adapted during training. But for evaluation where you don't update the parameters, this does not make a difference. The `model.eval()` command deactivates the Dropout modules of the model, such that a forwardpass of one input always produces the same output. Since the Dropout randomly drops some of the nodes this is not the case for models with active Dropout.  This explains the different results: `model.eval()` deactivates Dropout, while `model.freeze_model(True)` does not.
I hope this clarifies things.

**Comment by seahou3e** [comment]
@hSterz Thanks!
